docs/quickstart.md

```diff
@@ -0,0 +1,760 @@
+
+            def set(self, path, method, handler):
+                self.routes[f"{path}-{method}"] = handler
+
+            def get(self, path, method):
+                try:
+                    route = self.routes[f"{path}-{method}"]
+                except:
+                    print("Cannot route request to correct method")
+                    raise NotImplemented
+                return route
+
+
+        router = Router()
+        router.set(path="/hello", method="GET", handler=hello)
+        router.set(path="/hello/{name}", method="GET", handler=hello_name)
+
+
+        def lambda_handler(event, context):
+            path = event["resource"]
+            http_method = event["httpMethod"]
+            method = router.get(path=path, method=http_method)
+            return method(event=event)
+    ```
+
+=== "template.yaml"
+
+    ```yaml hl_lines="18-28"
+        AWSTemplateFormatVersion: "2010-09-09"
+        Transform: AWS::Serverless-2016-10-31
+        Description: >
+            Sample SAM Template for powertools-quickstart
+        Globals:
+            Function:
+            Timeout: 3
+        Resources:
+            HelloWorldFunction:
+            Type: AWS::Serverless::Function
+            Properties:
+                CodeUri: hello_world/
+                Handler: app.lambda_handler
+                Runtime: python3.9
+                Events:
+                    HelloWorld:
+                        Type: Api
+                        Properties:
+                        Path: /hello
+                        Method: get
+                    HelloWorldName:
+                        Type: Api 
+                        Properties:
+                        Path: /hello/{name}
+                        Method: get
+        Outputs:
+            HelloWorldApi:
+                Description: "API Gateway endpoint URL for Prod stage for Hello World function"
+                Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/"
+    ```
+In this example we added two methods `hello_name` and `hello` next to each other. We added `Router` class that allow us to register what method should be called when specific request comes. 
+We created instance of it and added configuration with mapping between processing methods and http request method. 
+In lambda handler we call router instance get method to fetch reference to processing method (`hello` or `hello_name`) that will handle the request. 
+Finally we run this method and return result to API Gateway.
+
+Thanks to this approach, we can simplify our infrastructure configuration since we added all API Gateway path under `HelloWorldFunction` event section. 
+Downside of current approach is that we need to add custom class and with every new method or path we want to serve we need to add explicit configuration. 
+Also, we need to understand API Gateway request event structure in order to decapsulate requested path and http method. 
```

Comment:
Hmm, can you elaborate on why it is a downside, or what problem it might cause? I assume you mean we need to know the directory structure of the event object (where we did not have error handling). It might need more clarification.

Reply:
I removed this part, after a second consideration, it makes no sense :-). I rewrote this section to this: 
`This approach allows us to simplify the configuration of our infrastructure since we have added all Gateway API paths in the `HelloWorldFunction` event section. We need to understand the internal structure of the API Gateway request events though, to deduce the requested path and the http method. This requires us to put additional engineering effort to provide proper error handling. Also, if we decide to use another event source for our lambda, since we are highly coupled it would require rewriting of our lambda handler to get the information we need.`
 Do you think it is accurate?